pkg/controller/certificates/internal/test/test.go

```diff
@@ -79,7 +79,11 @@ func MustCreateCryptoBundle(t *testing.T, crt *cmapi.Certificate, fixedClock *fa
 	return *c
 }
 
-func createCryptoBundle(crt *cmapi.Certificate, fixedClock *fakeclock.FakeClock) (*CryptoBundle, error) {
+func createCryptoBundle(originalCert *cmapi.Certificate, fixedClock *fakeclock.FakeClock) (*CryptoBundle, error) {
+	crt := originalCert.DeepCopy()
+	if crt.Spec.PrivateKey == nil {
+		crt.Spec.PrivateKey = &cmapi.CertificatePrivateKey{}
+	}
```

Comment:
What I really want is to create a `SetCertificateDefaults` function for Certificate and call that  everywhere immediately after a Certificate is created or after it has been retrieved from the API server.
Which would avoid having to check for nil everywhere.
I could do it in this PR or I could do it in a follow up.